File: core/src/galenscovell/oregontrail/processing/controls/GestureHandler.java
package galenscovell.oregontrail.processing.controls;

import com.badlogic.gdx.input.GestureDetector;
import galenscovell.oregontrail.ui.screens.GameScreen;

public class GestureHandler extends GestureDetector.GestureAdapter {
    private final GameScreen game;

    public GestureHandler(GameScreen game) {
        this.game = game;
    }

    @Override
    public boolean zoom (float initialDistance, float endDistance){
        // game.screenZoom(endDistance - initialDistance);
        return true;
    }

    @Override
    public boolean pan(float x, float y, float deltaX, float deltaY) {
        // game.screenPan(deltaX, deltaY);
        return true;
    }
}

File: core/src/galenscovell/hinterstar/processing/controls/GestureHandler.scala
package galenscovell.hinterstar.processing.controls

import com.badlogic.gdx.input.GestureDetector
import galenscovell.hinterstar.ui.screens.GameScreen


class GestureHandler(root: GameScreen) extends GestureDetector.GestureAdapter {
  private val gameScreen: GameScreen = root


  override def pan(x: Float, y: Float, deltaX: Float, deltaY: Float): Boolean = {
    gameScreen.actionPan(deltaX / 5)
    true
  }

  override def panStop(x: Float, y: Float, pointer: Int, button: Int): Boolean = {
    true
  }

  override def zoom(initialDistance: Float, endDistance: Float): Boolean = {
    val distance: Float = (initialDistance - endDistance) / 30000
    gameScreen.actionZoom(distance)
    true
  }
}
